src/components/TutorialView.jsx

```diff
@@ -108,13 +99,26 @@ class TutorialView extends React.Component {
     return currentStep >= maxSteps;
   }
 
+  handleStep(index) {
+    this.setState({
+      stepIndex: index
+    });
+  }
+
   render() {
     if (!this.state.loaded) {
       return null;
     }
+    const totalSteps = this.props.tutorial.steps.length || 0;
```

Comment:
Echoing my comments from earlier, if we're worried that `this.props.tutorial.steps.length` might be `undefined`, then it'd be best if we checked the existence of `this.props.tutorial.steps`. Otherwise, this line of code doesn't do much; if `steps` exists, then it should have `.length` of 0 or more (it's unlikely to be a non-Array), and if `step` _doesn't_ exist, then this line of code will crash anyhow.